src/refs.c

```diff
@@ -1130,6 +1130,18 @@ int git_reference__update_terminal(
 	return error;
 }
 
+const char *git_reference__commit_type(const git_commit *commit)
```

Comment:
@pks-t Fixed style issues and added a test. Have verified that test fails on revert of this fix.